tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+					log.FailOnError(err, "Failed to validate K8s versions")
+				}
+
+			})
+
+			// Step("SetClusterContext \"\" (setting it back to default)") is not needed
+			// since the next statement (in the next step) is anyway a sontect switch to source
+			// err := SetClusterContext("")
+			// log.FailOnError(err, "Failed to SetClusterContext to default cluster")
+
+		})
+
+		Step("Deploy the applications on Src cluster", func() {
+			log.InfoD("Deploy the applications on Src cluster")
+
+			Step("Deploy applications", func() {
+				log.InfoD("Deploy applications")
+
+				log.InfoD("Switching to source context")
+				err := SetSourceKubeConfig()
+				log.FailOnError(err, "Failed to switch to context to source cluster")
+
+				sourceClusterAppsContexts = make([]*scheduler.Context, 0)
+				for i := 0; i < Inst().GlobalScaleFactor; i++ {
+					taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+					appContexts = ScheduleApplications(taskName)
+					for _, appCtx := range appContexts {
+						appCtx.ReadinessTimeout = appReadinessTimeout
+						namespace := GetAppNamespace(appCtx, taskName)
+						// (sourceNamespaces, sourceClusterAppsContexts) will always correspoond
+						sourceNamespaces = append(sourceNamespaces, namespace)
+						sourceClusterAppsContexts = append(sourceClusterAppsContexts, appCtx)
+					}
+				}
+			})
+
+			Step("Validate applications", func() {
+				ValidateApplications(sourceClusterAppsContexts)
+
+				log.InfoD("Switching to default context")
+				err := SetClusterContext("")
+				log.FailOnError(err, "Failed to SetClusterContext to default cluster")
+			})
+
+			log.Warn("Waiting for 2 minutes, hoping for any Custom Resources to finish starting up.\nThis logic will have to be changed to polling in the future, as just randomly waiting, without verification can easily lead to errors. Make sure that any errors you're seeing is not due to this.")
+			time.Sleep(time.Minute * 2)
+
+		})
+
+		Step("Creating rules for backup", func() {
+			log.InfoD("Creating pre rule for deployed apps")
+			for i := 0; i < len(appList); i++ {
+				preRuleStatus, ruleName, err := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "pre")
+				log.FailOnError(err, "Creating pre rule for deployed apps failed")
+				dash.VerifyFatal(preRuleStatus, true, "Verifying pre rule for backup")
+
+				if ruleName != "" {
+					preRuleNameList = append(preRuleNameList, ruleName)
+				}
+			}
+			log.InfoD("Creating post rule for deployed apps")
+			for i := 0; i < len(appList); i++ {
+				postRuleStatus, ruleName, err := Inst().Backup.CreateRuleForBackup(appList[i], orgID, "post")
+				log.FailOnError(err, "Creating post rule for deployed apps failed")
+				dash.VerifyFatal(postRuleStatus, true, "Verifying Post rule for backup")
+				if ruleName != "" {
+					postRuleNameList = append(postRuleNameList, ruleName)
+				}
+			}
+		})
```

Comment:
If you are not using pre and post rule while taking the backup, then why have this step?

Reply:
The particular spec (elastic-search-crd) I'm using currently doesn't use pre or post rules, but if anyone adds a spec which does use it, then this will ensure that the backup/restore will be consistent. Right?